System/Console/Haskeline.hs

```diff
@@ -244,8 +245,7 @@ getPassword x = promptedInput
                                         $ runCommandLoop tops prefix loop
                                         $ Password [] x)
                     (\fops -> let h_in = inputHandle fops
-                              in bracketSet (hGetEcho h_in) (hSetEcho h_in) False
-                                  $ runMaybeT $ getLocaleLine fops)
+                              in withoutInputEcho h_in $ runMaybeT $ getLocaleLine fops)
```

Comment:
Honestly, no, that doesn't quite make sense to me. You say that this change would mean we could get rid of `inputHandle`, but that would obviate the need for https://github.com/judah/haskeline/pull/52#discussion_r98360675! So I'm a bit lost.